dotCMS/src/main/java/com/dotcms/graphql/datafetcher/ContentMapDataFetcher.java

```diff
@@ -131,16 +122,72 @@ private void enrichWithParsedRawFields(final Map<String, Object> hydratedMap) {
-                Collectors.toList()).get(0)).getOrNull();
+                        Collectors.toList()).get(0)).getOrNull();
 
         if(!isFieldRenderable(field)) {
             return contentlet.get(key);
         }
 
         return renderFieldValue(request, response, (String) contentlet.get(key), contentlet, field.variable());
     }
+
+    /**
+     * Attempts to retrieve a hydrated contentlet map, respecting an explicitly requested variant if present.
+     * <p>
+     * Behavior:
+     * <ul>
+     *   <li>If a variant is explicitly provided via {@link VariantAPI#VARIANT_KEY} in the request,
+     *       it is set on the contentlet.</li>
+     *   <li>If the variant is {@code DEFAULT}, no fallback is performed on error.</li>
+     *   <li>If no variant is provided, the contentlet will implicitly use the {@code DEFAULT} variant.</li>
+     *   <li>If a variant is set and fails to resolve (e.g., the contentlet has no version for it),
+     *       a fallback attempt is made using the {@code DEFAULT} variant.</li>
+     * </ul>
+     *
+     * @param request    the current HTTP request, used to extract the variant key if provided
+     * @param user       the user requesting the content, used for permissions and rendering
+     * @param contentlet the contentlet to hydrate
+     * @param render     whether renderable fields should be velocity-rendered
+     * @return a hydrated content map including field values (rendered if requested)
+     */
+    private Map<String, Object> getHydratedMapWithVariantFallback(
+            final HttpServletRequest request,
+            final User user,
+            final Contentlet contentlet,
+            final boolean render
+    ) {
+        final Object variantAttr = request.getAttribute(VariantAPI.VARIANT_KEY);
+
+        // Set variantId only if explicitly requested
+        if (UtilMethods.isSet(variantAttr)) {
+            final String variantName = variantAttr.toString();
+
+            // Avoid fallback loop if DEFAULT is already being used
+            assert VariantAPI.DEFAULT_VARIANT.name() != null;
+            if (VariantAPI.DEFAULT_VARIANT.name().equals(variantName)) {
+                contentlet.setVariantId(variantName);
+                return ContentletUtil.getContentPrintableMap(user, contentlet, false, render);
+            }
+
+            try {
+                contentlet.setVariantId(variantName);
+                return ContentletUtil.getContentPrintableMap(user, contentlet, false, render);
+            } catch (DotStateException e) {
+                Logger.debug(this, () -> String.format(
+                        "Variant '%s' not found for contentlet '%s'. Falling back to DEFAULT variant.",
+                        variantName, contentlet.getIdentifier()
+                ));
+
+                contentlet.setVariantId(VariantAPI.DEFAULT_VARIANT.name());
+                return ContentletUtil.getContentPrintableMap(user, contentlet, false, render);
```

Comment:
It all looks good, but I'm concerned that you're modifying a contentlet, and whenever we do that, we need to be mindful of whether we're affecting the in-cache copy. The good practice is always creating a copy and modifying the copy so that the element in cache remains unmodified